### PR TITLE
Reindex Test, cover a few more test cases, additional config options.

### PR DESCRIPTION
**Pull request by markrmiller**
Adds a basic test focused on reindexing, covers a few more test cases: more replicas and shards, different shard count on primary and secondary dc, allows consumer group to be configured, allows crossdc zk prop file location to be configured.

### CODE REVIEW

**Review by anshumg — APPROVED** [review]

File: crossdc-consumer/src/main/java/org/apache/solr/crossdc/consumer/KafkaCrossDcConsumer.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public KafkaCrossDcConsumer(KafkaCrossDcConf conf) {
 
     kafkaConsumerProp.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, conf.getBootStrapServers());
 
-    kafkaConsumerProp.put(ConsumerConfig.GROUP_ID_CONFIG, "group_1"); // TODO
+    kafkaConsumerProp.put(ConsumerConfig.GROUP_ID_CONFIG, conf.getGroupId()); // TODO
```
Comment (anshumg) [inline crossdc-consumer/src/main/java/org/apache/solr/crossdc/consumer/KafkaCrossDcConsumer.java:52]: We can get rid of the `//TODO` here?

**Review by anshumg — CHANGES_REQUESTED** [review]
We need to handle empty/malformed/missing groupId, topicName, etc. here.
I personally feel it doesn't make sense to have a default but I'm open to a counter argument.

**Comment by markrmiller** [comment]
Thanks for the review. Meant to get my update up today, but got sidetracked on another item. Will get up tomorrow.

**Comment by markrmiller** [comment]
> We need to handle empty/malformed/missing groupId, topicName, etc. here. 

Do we? That means copying and duplicating and evolving the rules from/with Kafka when Kafka already handles it for us. We pass the input to Kafka, it validates it, it spits out an error message.

> I personally feel it doesn't make sense to have a default but I'm open to a counter argument.

The counter is that it's required to pass a group id so you need some default.

**Comment by markrmiller** [comment]
> The counter is that it's required to pass a group id so you need some default.

Though, an empty string is a valid default to use as far as I know.

**Comment by markrmiller** [comment]
Okay, I was reading that as kind of two different comments, and it just jelled for me. You are saying you think we should make groupid required in our config?

I guess I would say that feels a little unnecessary given it doesn't necessarily always have value for a setup and if it does a user would know that they want that value and be able to use it, and someone trying things out or that didn't need it wouldn't have to add one more bit of config that they may or may not understand initially. To pick a reasonable group id, you really have to understand something about Kafka and what it is and if you understand that you know whether you want a group id or not and if you don't know all that, it's just a seemingly not very relevant addition to the current very simple required config for someone getting started.

**Comment by markrmiller** [comment]
For the moment I just put up an update I already had together. As far as requiring group id, I guess I see it the opposite way, eg what's a good reason to require it? Mainly from the stand point that minimal required configuration to get going is easier to get started with and so my intent with making so little configurable to start was to try and keep what was required minimal and that what was possible on that either kind of minimally necessary  and/or offer a way to generically take and pass on anything Kafka can accept in a way that that we don't own each option as well, just what we make required.

**Pull request merged**